KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift

```diff
@@ -342,42 +367,61 @@ final class KarhooLoyaltyPresenter: LoyaltyPresenter {
         return currentMode == .burn ? viewModel.burnAmount : 0
     }
     
-    private func getSubtitleText() -> String {
+    private func getEarnText() -> String {
         switch currentMode {
         case .none:
             return ""
         case .burn:
-            return String(format: NSLocalizedString(UITexts.Loyalty.pointsBurnedForTrip, comment: ""), "\(viewModel?.burnAmount ?? 0)")
+            return viewModel?.canEarn ?? false ?
```

Comment:
Whoa, what a bunch of code! Break it and create a local constant, so another developer can follow the flow easily.